addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/BaseHttpSender.java

```diff
@@ -69,11 +79,63 @@
             HttpRequestConfig.builder().setFollowRedirects(true).build();
 
     private final List<HttpSenderListener> listeners;
+    private final Map<HttpSender, T1> contexts;
 
+    @SuppressWarnings("unchecked")
     protected BaseHttpSender() {
         listeners = new ArrayList<>();
+        contexts =
+                Collections.synchronizedMap(
+                        new ReferenceIdentityMap(
+                                AbstractReferenceMap.WEAK, AbstractReferenceMap.HARD));
+    }
+
+    @Override
+    public Object saveState() {
+        synchronized (contexts) {
+            return Map.of(
```

Comment:
We could use an instance of an `HttpSenderState` (or similar) class instead of a map? That way we wouldn't have to keep a track of the types of the values in the map.

Reply:
We can't use classes defined in/by the add-on, that would lead to class cast exceptions (as the new add-on has one class and the old another). It could be changed to use serialization though.

Reply:
That makes sense.